docs/task.md

```diff
@@ -0,0 +1,391 @@
+# Task Schema
+
+## Task Properties
+
+| Property | Type | Required | Nullable |
+|----------|------|----------|----------|
+| [steps](#steps) | `step[]` | Required  | No |
+| [stepTimeout](#steptimeout) | `int` | Optional  | Yes |
+| [totalTimeout](#totaltimeout) | `int` | Optional | Yes |
+| [secrets](#secrets) | `secret[]` | Optional | Yes |
+| [networks](#networks) | `network[]` | Optional | Yes |
+| [envs](#envs) | `string[]` | Optional | Yes |
+| [workingDirectory](#workingdirectory) | `string` | Optional | Yes |
+| [version](#version) | `string` | Optional | Yes |
+
+## steps
+
+An array of [step](#step) objects.
+
+* Required
+* Type: `step[]`
+
+## stepTimeout
+
+A [step's](#step) maximum execution time in seconds. This property defaults all [steps'](#steps) [timeout](#timeout) properties. A [step](#step) can override this property via [timeout](#timeout).
+
+* Optional
+* Type: `int`
+
+## totalTimeout
+
+The total execution execution time of the [task](#task) in seconds.
+
+* Optional
+* Type: `int`
+
+## secrets
+
+An array of [secret](#secret) objects.
+
+* Optional
+* Type: `secret[]`
+
+## networks
+
+An array of [network](#network) objects.
+
+* Optional
+* Type: `network[]`
+
+## envs
+
+Default environment variables which are applied to every [step](#step) in the format of `VARIABLE=value`.
+
+* Optional
+* Type: `string[]`
+
+## workingDirectory
+
+Specifies the working directory of the container during runtime. If specified on a [task](#task), it defaults all of the [steps'](#step) `workingDirectory` properties. If specified on a [step](#step), it will override the property provided by the [task](#task).
+
+* Optional
+* Type: `string`
+
+## version
+
+The version of the [task](#task). If unspecified, defaults to the latest version.
+
+* Optional
+* Type: `string`
+
+### step
+
+An object with the following properties:
+
+| Property | Type | Required | Nullable |
+|----------|------|----------|----------|
+| [id](#id) | `string` | Optional | Yes |
```

Comment:
What does Nullable mean? Especially for int or bool type. I think we probably need a default value column.

Reply:
Removed, added default value